File: Learnify.Service/Implementation/EnrollmentService.cs
using Learnify.Domain.DTO;
using Learnify.Domain.Entities;
using Learnify.Repository.Interfaces;
using Learnify.Service.Interfaces;
using Learnify.Service.Mappers;

namespace Learnify.Service.Implementation
{
    public class EnrollmentService : IEnrollmentService
    {
        private readonly IEnrollmentRepository enrollmentRepository;

        public EnrollmentService(IEnrollmentRepository enrollmentRepository)
        {
            this.enrollmentRepository = enrollmentRepository;
        }

        public Task DeleteEnrollment(Guid enrollmentId)
        {
            var enrollment = enrollmentRepository.Get(enrollmentId);

            enrollmentRepository.Delete(enrollment);

            return Task.CompletedTask;
        }

        public Enrollment GetEnrollment(Guid enrollmentId)
        {
            var enrollment = enrollmentRepository.Get(enrollmentId);

            return enrollment;
        }

        public List<Enrollment> GetEnrollments(EnrollmentRequest request)
        {
            var enrollments = enrollmentRepository.GetEnrollments(request);

            return enrollments.ToList();
        }

        public Task InsertEnrollment(InsertEnrollment insertEnrollment)
        {
            var enrollment = EnrollmentMapper.CopyFromDto(insertEnrollment);

            enrollmentRepository.Insert(enrollment);

            return Task.CompletedTask;
        }

        public Task UpdateEnrollment(UpdateEnrollment updateEnrollment)
        {
            var enrollment = EnrollmentMapper.CopyFromUpdateDto(updateEnrollment);

            enrollmentRepository.Update(enrollment);

            return Task.CompletedTask;
        }
    }
}

File: Learnify.UI/src/services/EnrollmentService.js
import { axios_config } from '../axios/axios_config';

const endpoint = '/enrollments';

export const getEnrollments = async (request) => {
    try {
        const response = await axios_config.post(`${endpoint}/get`, request);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getEnrollmentById = async (enrollmentId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${enrollmentId}`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const deleteEnrollment = async (enrollmentId) => {
    try {
        const response = await axios_config.delete(`${endpoint}/${enrollmentId}`);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const insertEnrollment = async (insertEnrollment) => {
    try {
        const response = await axios_config.post(`${endpoint}/insert`, insertEnrollment);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const updateEnrollment = async (updateEnrollment) => {
    try {
        const response = await axios_config.put(endpoint, updateEnrollment);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};
